code/modules/antagonists/traitor/objectives/demoralise_graffiti.dm

```diff
@@ -0,0 +1,335 @@
+	var/obj/effect/decal/cleanable/traitor_rune/rune
+
+/datum/traitor_objective/demoralise/graffiti/generate_ui_buttons(mob/user)
+	var/list/buttons = list()
+	if (!obtained_spray)
+		buttons += add_ui_button("", "Pressing this will materialize a syndicate spraycan in your hand.", "wifi", "summon_gear")
+	else
+		buttons += add_ui_button("[demoralised_crew_events] / [demoralised_crew_required] propagandised", "This many crew have been exposed to propaganda, out of a required [demoralised_crew_required].", "wifi", "none")
+	return buttons
+
+/datum/traitor_objective/demoralise/graffiti/ui_perform_action(mob/living/user, action)
+	. = ..()
+	switch(action)
+		if ("summon_gear")
+			if (obtained_spray)
+				return
+
+			obtained_spray = TRUE
+			var/obj/item/traitor_spraycan/spray = new(user.drop_location())
+			user.put_in_hands(spray)
+			spray.balloon_alert(user, "the spraycan materializes in your hand")
+
+			RegisterSignal(spray, COMSIG_PARENT_QDELETING, .proc/on_spray_destroyed)
+			RegisterSignal(spray, COMSIG_TRAITOR_GRAFFITI_DRAWN, .proc/on_rune_complete)
+
+/**
+ * Called when the spray can is deleted.
+ * If it's already been expended we don't care, if it hasn't you just made your objective impossible.area
+ *
+ * Arguments
+ * * spray - the spraycan which was just deleted
+ */
+/datum/traitor_objective/demoralise/graffiti/proc/on_spray_destroyed()
+	SIGNAL_HANDLER
+	// You fucked up pretty bad if you let this happen
+	if (!rune)
+		fail_objective()
+
+/**
+ * Called when you managed to draw a traitor rune.
+ * Sets up tracking for objective progress, and unregisters signals for the spraycan because we don't care about it any more.
+ *
+ * Arguments
+ * * drawn_rune - graffiti 'rune' which was just drawn.
+ */
+/datum/traitor_objective/demoralise/graffiti/proc/on_rune_complete(atom/spray, obj/effect/decal/cleanable/traitor_rune/drawn_rune)
+	SIGNAL_HANDLER
+	rune = drawn_rune
+	UnregisterSignal(spray, COMSIG_PARENT_QDELETING)
+	UnregisterSignal(spray, COMSIG_TRAITOR_GRAFFITI_DRAWN)
+	RegisterSignal(drawn_rune, COMSIG_PARENT_QDELETING, .proc/on_rune_destroyed)
+	RegisterSignal(drawn_rune, COMSIG_DEMORALISING_EVENT, .proc/on_mood_event)
+
+/**
+ * Called when your traitor rune is destroyed. If you haven't suceeded by now, you fail.area
+ *
+ * Arguments
+ * * rune - the rune which just got destroyed.
+ */
+/datum/traitor_objective/demoralise/graffiti/proc/on_rune_destroyed(obj/effect/decal/cleanable/traitor_rune/rune)
+	SIGNAL_HANDLER
+	fail_objective()
+	UnregisterSignal(rune, COMSIG_PARENT_QDELETING)
+	UnregisterSignal(rune, COMSIG_DEMORALISING_EVENT)
+
+/datum/traitor_objective/demoralise/graffiti/on_success(mob/owner)
+	. = ..()
+	UnregisterSignal(rune, COMSIG_PARENT_QDELETING)
+	UnregisterSignal(rune, COMSIG_DEMORALISING_EVENT)
+
+/**** Syndicate spraycan ****/
+
+// Extending the existing spraycan item was more trouble than it was worth, I don't want or need this to be able to draw arbitrary shapes.
+/obj/item/traitor_spraycan
```

Comment:
Almost nothing, which is why I elected not to make it a child of spraycan. 
Really all I actually want is "an item which has an icon which looks like a spraycan" and none of the functionality because the ability to draw other arbitrary things with it just made its usage confusing. This spraycan only draws your objective and nothing else.
I think the only part I copied was the `suicide_act` (refactored to remove a bunch of checks redundant for this item) with the assumption that someone will try to do it some day.